### DISCUSSION TITLE
Case Study – Next.js 13

### DISCUSSION THREAD

**Discussion by ericclemmons** [ericclemmons/project-pontis#6] [Ideas]
The purpose of this case study is to document the _cold-start_ experience of Next.js + Tailwind and identify how #2 could apply:

> https://nextjs.org/docs/getting-started

**EDIT: I meant to use the new `app` docs:**
> https://beta.nextjs.org/docs/installation

**Comment by ericclemmons** [comment]
To first get started, I made an `examples/` folder.

If there were a Next.js Bridge, it could have a `New Next.js App...` prompt:
> <img width="465" alt="Screen Shot 2022-11-14 at 2 54 32 PM" src="https://user-images.githubusercontent.com/15182/201763673-32f27a0d-4a02-4fa0-bd73-0d4992879242.png">

Instead, I'm going into the Terminal to run `pnpm create next-app --typescript`.

**Edit: I'm instead running `pnpm create next-app --experimental-app`**

**Comment by ericclemmons** [comment]
When inspecting an element (via right-click, or clicking on a node in the Elements panel), I'd like to the bridge:

1. Open the file, if it's not already open
2. Put the cursor (or select?) the node in the code

> <img width="744" alt="Screen Shot 2022-11-14 at 3 04 16 PM" src="https://user-images.githubusercontent.com/15182/201765327-36927238-4562-43ac-8a08-e2a0c977737e.png">
<img width="1440" alt="Screen Shot 2022-11-14 at 3 03 50 PM" src="https://user-images.githubusercontent.com/15182/201765339-51e31e1a-8ecd-4638-99af-4dba0ccb1162.png">

**Comment by ericclemmons** [comment]
Creating a new `page.js` has boilerplate.  What if the right-click menu said `New Page...`?

> <img width="370" alt="Screen Shot 2022-11-14 at 3 06 59 PM" src="https://user-images.githubusercontent.com/15182/201765747-326e8606-ccd5-4c79-b77e-47fea0b3d952.png">

**Reply by ericclemmons** [reply]
Or perhaps a dedicated sidebar "Routes" panel with a welcome view:
> ![](https://code.visualstudio.com/assets/api/ux-guidelines/examples/welcome-view.png)
> https://code.visualstudio.com/api/ux-guidelines/views#welcome-views

**Comment by ericclemmons** [comment]
Adding TailwindCSS was a breeze, but it required closing the server, adding dependencies, then restarting the server:
> https://beta.nextjs.org/docs/styling/tailwind-css#installing-tailwind

I suppose that isn't a big deal, but if there's a formula for it, why not have it installable somehow?
> <img width="723" alt="Screen Shot 2022-11-14 at 3 24 55 PM" src="https://user-images.githubusercontent.com/15182/201768738-a3e3a027-8eab-4561-afeb-82616ae54761.png">

**Comment by ericclemmons** [comment]
Doing things _visually_ is still PITA of trial-and-error.

> <img width="1440" alt="Screen Shot 2022-11-14 at 3 23 06 PM" src="https://user-images.githubusercontent.com/15182/201769945-1979bc71-97ac-4cc4-a3f8-4ace5a133ac0.png">

On one hand, I'd love to have a sidebar (ala codelift.app) to give visual controls for Tailwind classes that directly manipulated the code.

> https://code.visualstudio.com/api/ux-guidelines/sidebars

On the other, changing code is the _result_ of the UI looking "right", so perhaps a Chrome extension/panel is better suited for controlling this?

> https://developer.chrome.com/docs/extensions/reference/devtools_panels/#examples

**Reply by cdonohue** [reply]
The interesting part about this is figuring out (supporting) the many different ways visual elements are styled and providing a way to write back to the source in the right location. Off the top of my head:
- Static style sheet
- `<style>` tags
- Compiled styles (CSS in JS, PostCSS, etc)

We'd probably start with these and abstract away the common pieces to allow others to build the next example that we haven't thought of at this moment 😄.

**Comment by ericclemmons** [comment]
Also see:

https://twitter.com/delba_oliveira/status/1592209522281439232?s=46&t=zWsM6ayyDaWmEuPMCysntw
